### ISSUE TITLE
Speed up travis-ci build

### ISSUE THREAD

**Issue by GiovanniBussi** [plumed/plumed2#286]
OSX builds are too just slow. We could remove some of them (e.g. keeping only one with homebrew and one with macports). Or, better, we could have some of the tests that are only executed in special conditions. Notice that exiting prematurely is not a solution, since osx builds waste a lot of time in the queue (before even starting).

A possible solution is:
- Comment out a number of test from `.travis.yml` with some special string (e.g. `#@#` at the beginning of the line)
- Whenever there is a commit, check if there is a special string in the message (e.g. `[fulltest]`).
- If so, edit the `.travis.yml` with sed removing the extra comment and then use `git add .travis.yml` and git commit --amend` to change the file.
- Push the resulting commit to a fake branch on [plumedbot fork](https://github.com/plumedbot/plumed2) (which is writable from travis-ci).
- Tests will be executed by the `plumedbot` user, thus will not even consume the 5 running jobs on the `plumed` organization.

If this works, we might even add more hidden jobs, e.g. to test more recent compilers. Then we could perform them only when we know we are making critical changes or close to any release.

**Comment by carlocamilloni** [comment]
I think that OSX builds are not very important, all of us are using OSX for development so it is very well tested, I think it would be fine to keep only two as you wrote.
About the two levels of testing that could be nice, gromacs uses something like this. In particular for all the commits related to documentation there is no need to have all the tests done.

**Comment by GiovanniBussi** [comment]
I actually did it in a much simpler way exploiting conditional builds (see branch fix-286). In this manner it is however not possible to parse the log message. The only easily accessible information is the branch name. The change I did works as follows:
- If branch name is test-*, all tests are done
- if branch name is anything else, only a few tests are done

I am planning to also revise a bit the test list, in particular I would keep as "always done":
- MPI and debug options, building the documentation every time (that is: `[makedoc]` will be ignored (a mixture of present jobs 1 and 3)
- No MPI and debug options (now job 2)
- CPPCHECK stuff (now job 4)
- Normal MPI setup with optimization flags and system blas+lapack (now job 5 + MPI)
- A single Macports build in order to check that packaging is correct.

The main change would be that **documentation will be always built**, which I think is useful because it also means that coverage tests are always up to date.

Let me know if this is ok!

Giovanni
